addons/src/ao_tasks.c

```diff
@@ -556,15 +565,22 @@ static void create_task(AoTasks *t, GeanyDocument *doc, gint line, const gchar *
 }
 
 
-static void update_tasks_for_doc(AoTasks *t, GeanyDocument *doc)
+static gboolean update_tasks_for_doc_idle_cb(gpointer data)
 {
+	AoTasksUpdateTasksForDocArguments *arguments = data;
+	AoTasks *t = arguments->t;
+	GeanyDocument *doc = arguments->doc;
```

Comment:
Needs to check valid since its an unknown period and unknown actions before it happens (isn't asynchronous fun!)

Reply:
I don't know if there is more than checking `data` for `NULL`. BUt at least I've added this.

Reply:
I was thinking of DOCVALID()